prost/src/message.rs

```diff
@@ -145,12 +145,22 @@ pub trait Message: Send + Sync {
             WireType::LengthDelimited,
             self,
             &mut buf,
-            DecodeContext::default(),
+            DecodeContext::new(Self::recursion_limit()),
         )
     }
 
     /// Clears the message, resetting all fields to their default.
     fn clear(&mut self);
+
+    /// The recursion limit for decoding protobuf messages.
+    ///
+    /// Defaults to 100. Can be customized in your build.rs or by using the no-recursion-limit crate feature.
+    fn recursion_limit() -> u32
```

Comment:
Is this a breaking change? I feel like it is not, because you add a default implementation. However rust rules for breaking change are not always logical. Could you verify this with `cargo-semver-checks`?